ansible/templates/minigraph_link_meta.j2

```diff
@@ -1,6 +1,25 @@
-{% if 'dualtor' in topo %}
+{% set link_metadata_defined = False %}
+
+{% if 'dualtor' in topo or
+      (macsec_card is defined and macsec_card == True and 't2' in topo) %}
```

Comment:
@vdahiya12 unable to understand macsec and dualtor relationship with link_metadata_defined.

Reply:
@prgeor actually if you see the code, there are lots of places where try to add and close this tag

```
<LinkMetadataDeclaration>
    <Link xmlns:a="http://schemas.datacontract.org/2004/07/Microsoft.Search.Autopilot.Evolution">
 ```
 what my change also addresses is that there is a unified way for all macsec/dualtor/autoneg etc. or anything incoming infuture to check if this above declaration is needed. 
 Once this is determined, it's easy to just place these top level checks and add the linkmetadatadeclartion tag only once.